lms/djangoapps/badges/backends/tests/test_badgr_backend.py

```diff
@@ -73,121 +82,110 @@ def check_headers(self, headers):
-        """
-        Verify slug generation is working as expected. If this test fails, the algorithm has changed, and it will cause
-        the handler to lose track of all badges it made in the past.
-        """
-        self.assertEqual(
-            self.handler.course_slug('honor'),
-            'edxcourse_testtest_run_honor_fc5519b'
-        )
-        self.assertEqual(
-            self.handler.course_slug('verified'),
-            'edxcourse_testtest_run_verified_a199ec0'
-        )
-
     def test_get_headers(self):
         """
         Check to make sure the handler generates appropriate HTTP headers.
         """
-        self.check_headers(self.handler.get_headers())
+        self.check_headers(self.handler._get_headers())
 
     @patch('requests.post')
     def test_create_badge(self, post):
         """
         Verify badge spec creation works.
         """
-        self.handler.create_badge('honor')
+        self.handler._create_badge(self.badge_class)
         args, kwargs = post.call_args
         self.assertEqual(args[0], 'https://example.com/v1/issuer/issuers/test-issuer/badges')
-        self.assertEqual(kwargs['files']['image'][0], BadgeImageConfiguration.objects.get(mode='honor').icon.name)
+        self.assertEqual(kwargs['files']['image'][0], self.badge_class.image.name)
         self.assertIsInstance(kwargs['files']['image'][1], ImageFieldFile)
         self.assertEqual(kwargs['files']['image'][2], 'image/png')
         self.check_headers(kwargs['headers'])
         self.assertEqual(
             kwargs['data'],
             {
-                'name': 'Badged',
-                'slug': 'edxcourse_testtest_run_honor_fc5519b',
-                'criteria': 'https://edx.org/courses/edX/course_test/test_run/about',
-                'description': 'Completed the course "Badged" (honor, 2015-05-19 - 2015-05-20)',
+                'name': 'Test Badge',
+                'slug': '15bb687e0c59ef2f0a49f6838f511bf4ca6c566dd45da6293cabbd9369390e1a',
+                'criteria': 'https://example.com/syllabus',
+                'description': "Yay! It's a test badge.",
             }
         )
 
-    def test_self_paced_description(self):
-        """
-        Verify that a badge created for a course with no end date gets a different description.
-        """
-        self.course.end = None
-        self.assertEqual(BadgeHandler.badge_description(self.course, 'honor'), 'Completed the course "Badged" (honor)')
-
     def test_ensure_badge_created_cache(self):
         """
         Make sure ensure_badge_created doesn't call create_badge if we know the badge is already there.
         """
-        BadgeHandler.badges['edxcourse_testtest_run_honor_fc5519b'] = True
-        self.handler.create_badge = Mock()
-        self.handler.ensure_badge_created('honor')
-        self.assertFalse(self.handler.create_badge.called)
+        BadgrBackend.badges.append('15bb687e0c59ef2f0a49f6838f511bf4ca6c566dd45da6293cabbd9369390e1a')
```

Comment:
I know it's repetitive, but in texts I like to use string literals in tests like this where possible, since deriving new strings and changing their values can have unanticipated consequences, and that is precisely the sort of thing we're trying to test for.